src/test/java/engineering/swat/watch/SingleFileTests.java

```diff
@@ -0,0 +1,94 @@
+package engineering.swat.watch;
+
+import static org.awaitility.Awaitility.await;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.attribute.FileTime;
+import java.time.Instant;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class SingleFileTests {
+    private TestDirectory testDir;
+
+    @BeforeEach
+    void setup() throws IOException {
+        testDir = new TestDirectory();
+    }
+
+    @AfterEach
+    void cleanup() throws IOException {
+        if (testDir != null) {
+            testDir.close();
+        }
+    }
+
+    @BeforeAll
+    static void setupEverything() {
+        Awaitility.setDefaultTimeout(TestHelper.NORMAL_WAIT);
+    }
+
+    @Test
+    void singleFileShouldNotTriggerOnOtherFilesInSameDir() throws IOException, InterruptedException {
+        var target = testDir.getTestFiles().get(0);
+        var seen = new AtomicBoolean(false);
+        var others = new AtomicBoolean(false);
+        var watchConfig = Watcher.watch(target, WatchScope.PATH_ONLY)
+            .onEvent(ev -> {
+                if (ev.calculateFullPath().equals(target)) {
+                    seen.set(true);
+                }
+                else {
+                    others.set(true);
+                }
+            });
+        try (var watch = watchConfig.start()) {
+            for (var f : testDir.getTestFiles()) {
+                if (!f.equals(target)) {
+                    Files.writeString(f, "Hello");
+                }
+            }
+            Thread.sleep(TestHelper.SHORT_WAIT.toMillis());
+            Files.writeString(target, "Hello world");
+            await("Single file does trigger")
+                .pollDelay(TestHelper.NORMAL_WAIT.minusMillis(10))
+                .failFast("No others should be notified", others::get)
+                .untilTrue(seen);
+        }
+    }
+
+    @Test
+    void singleFileThatMonitorsOnlyADirectory() throws IOException, InterruptedException {
+        var target = testDir.getTestDirectory();
+        var seen = new AtomicBoolean(false);
+        var others = new AtomicBoolean(false);
+        var watchConfig = Watcher.watch(target, WatchScope.PATH_ONLY)
+            .onEvent(ev -> {
+                if (ev.calculateFullPath().equals(target)) {
+                    seen.set(true);
+                }
+                else {
+                    others.set(true);
+                }
+            });
+        try (var watch = watchConfig.start()) {
+            for (var f : testDir.getTestFiles()) {
+                if (!f.equals(target)) {
+                    Files.writeString(f, "Hello");
```

Comment:
I didn't realize a write to a file (`CREATED` and `DELETED` but not `MODIFIED`)? in a directory triggers an event for that directory. Maybe this can be documented in `WatchScope`.

Reply:
I don't understand this question perse, can you suggest an improvement for the documentation? maybe I understand the point that way?

Reply:
I've updated the documentation a bit to explain what I meant.

Reply:
ok, I changed it a bit, but thanks.